build/common.go

```diff
@@ -0,0 +1,286 @@
+		return nil
+	}
+
+	if err := sh.RunV("go", "tool", "cover", "-html=coverage/backend.out", "-o", "coverage/backend.html"); err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// Lint audits the source style
+func Lint() error {
+	return sh.RunV("golangci-lint", "run", "./...")
+}
+
+// Format formats the sources.
+func Format() error {
+	if err := sh.RunV("gofmt", "-w", "."); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Clean cleans build artifacts, by deleting the dist directory.
+func Clean() error {
+	err := os.RemoveAll("dist")
+	if err != nil {
+		return err
+	}
+
+	err = os.RemoveAll("coverage")
+	if err != nil {
+		return err
+	}
+
+	err = os.RemoveAll("ci")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkLinuxPtraceScope() {
+	ptracePath := "/proc/sys/kernel/yama/ptrace_scope"
+	byteValue, err := readFileBytes(ptracePath)
+	if err != nil {
+		fmt.Printf("Unable to read ptrace_scope\n")
+	}
+	val := strings.TrimSpace(string(byteValue[:]))
+	if "0" != val {
+		fmt.Printf("WARNING: \n")
+		fmt.Printf("ptrace_scope set to value other than 0 (currenlty:%s), this might prevent debugger from connecting\n", val)
+		fmt.Printf("try writing \"0\" to %s\n", ptracePath)
+		fmt.Printf("Set ptrace_scope to 0? y/N (default N)\n")
+
+		scanner := bufio.NewScanner(os.Stdin)
+		if scanner.Scan() {
+			if scanner.Text() == "y" || scanner.Text() == "Y" {
+				// if err := sh.RunV("echo", "0", "|", "sudo", "tee", ptracePath); err != nil {
+				// 	return // Error?
+				// }
+				fmt.Printf("TODO, run: echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope\n")
+			} else {
+				fmt.Printf("Did not write\n")
+			}
+		}
+	}
+}
+
+// Debugger makes a new debug build and attaches dlv (go-delve).
+func Debugger() error {
+	// Debug build
+	b := Build{}
+	mg.Deps(b.Debug)
+
+	// 1. kill any running instance
+	exeName := getExecutableName(runtime.GOOS, runtime.GOARCH)
+
+	// Kill any running processs
+	_ = killAllPIDs(findRunningPIDs(exeName))
+	_ = sh.RunV("pkill", "dlv")
+
+	if runtime.GOOS == "linux" {
+		checkLinuxPtraceScope()
+	}
+
+	// Wait for grafana to start plugin
+	for i := 0; i < 20; i++ {
+		pids := findRunningPIDs(exeName)
+		if len(pids) > 1 {
+			return fmt.Errorf("multiple instances already running")
+		}
+		if len(pids) > 0 {
+			pid := strconv.Itoa(pids[0])
+			log.Printf("Running PID: %s", pid)
+
+			// dlv attach ${PLUGIN_PID} --headless --listen=:${PORT} --api-version 2 --log
+			if err := sh.RunV("dlv",
+				"attach",
+				pid,
+				"--headless",
+				"--listen=:3222",
+				"--api-version", "2",
+				"--log"); err != nil {
+				return err
+			}
+			fmt.Printf("dlv finished running (%s)\n", pid)
+			return nil
+		}
+
+		log.Printf("waiting for grafana to start: %s...", exeName)
+		time.Sleep(250 * time.Millisecond)
+	}
+	return fmt.Errorf("could not find process: %s, perhaps grafana is not running?", exeName)
+}
+
+// Default configures the default target.
+var Default = BuildAll
```

Comment:
From my testing, importing the `Default` target doesn't work, it has to be in Magefile.go.
```suggestion
```